src/Middleware/Rewrite/src/Internal/IISUrlRewrite/ServerVariables.cs

```diff
@@ -19,54 +19,79 @@ public static class ServerVariables
         /// <returns>The matching <see cref="PatternSegment"/></returns>
         public static PatternSegment FindServerVariable(string serverVariable, ParserContext context, UriMatchPart uriMatchPart)
         {
+            Func<PatternSegment> fallback = default;
```

Comment:
This is a small a breaking change if one of these variables return a different value than the fallback did and there is no way to turn it off. We may need to add an option or something to not use IISServerVariables.

Reply:
True.

Going up the layers, I think we can add an optional boolean parameter to the [`AddIISUrlRewrite`](https://github.com/aspnet/AspNetCore/blob/b0659f660b6e8151a10da1c6c16310ec0101c380/src/Middleware/Rewrite/src/IISUrlRewriteOptionsExtensions.cs) extension methods and let it flow down to `ServerVariables.FindServerVariable`.

Reply:
Rook a shot at introducing a way to opt out in [`2e90d8f3`](https://github.com/aspnet/AspNetCore/commit/2e90d8f3c9b71834196f854ca3f6d64ad48ef7fd)